ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/QuestionAnsweringModel.java

```diff
@@ -47,50 +63,230 @@ public void warmUp(Predictor predictor, String modelId, MLModelConfig modelConfi
+    }
+
+    private boolean isSentenceHighlightingModel() {
+        return modelConfig != null && SENTENCE_HIGHLIGHTING_TYPE.equalsIgnoreCase(modelConfig.getModelType());
+    }
+
+    private ModelTensorOutput predictStandardQA(String question, String context) throws TranslateException {
         Input input = new Input();
         input.add(question);
         input.add(context);
-        output = getPredictor().predict(input);
-        tensorOutputs.add(parseModelTensorOutput(output, null));
-        return new ModelTensorOutput(tensorOutputs);
+
+        try {
+            Output output = getPredictor().predict(input);
+            ModelTensors tensors = parseModelTensorOutput(output, null);
+            return new ModelTensorOutput(List.of(tensors));
+        } catch (Exception e) {
+            log.error("Error processing standard QA model prediction", e);
+            throw new TranslateException("Failed to process standard QA model prediction", e);
+        }
+    }
+
+    private ModelTensorOutput predictSentenceHighlightingQA(String question, String context) throws TranslateException {
+        SentenceHighlightingQATranslator translator = (SentenceHighlightingQATranslator) getTranslator(PYTORCH_ENGINE, this.modelConfig);
+
+        try {
+            List<Map<String, Object>> allHighlights = new ArrayList<>();
+
+            // Process initial chunk
+            processInitialChunk(question, context, allHighlights);
+
+            // Process overflow chunks if any
+            processOverflowChunks(question, context, translator, allHighlights);
+
+            return createHighlightOutput(allHighlights);
+        } catch (Exception e) {
+            log.error("Error processing sentence highlighting model prediction", e);
+            throw new TranslateException("Failed to process chunks for sentence highlighting", e);
+        }
+    }
+
+    private void processInitialChunk(String question, String context, List<Map<String, Object>> allHighlights) throws TranslateException {
+        Input initialInput = new Input();
+        initialInput.add(MLInput.QUESTION_FIELD, question);
+        initialInput.add(MLInput.CONTEXT_FIELD, context);
+        initialInput.add(HIGHLIGHTING_MODEL_CHUNK_NUMBER_KEY, HIGHLIGHTING_MODEL_INITIAL_CHUNK_NUMBER_STRING);
+
+        List<Output> initialOutputs = getPredictor().batchPredict(List.of(initialInput));
+        for (Output output : initialOutputs) {
+            ModelTensors tensors = parseModelTensorOutput(output, null);
+            allHighlights.addAll(extractHighlights(tensors));
+        }
+    }
+
+    private void processOverflowChunks(
+        String question,
+        String context,
+        SentenceHighlightingQATranslator translator,
+        List<Map<String, Object>> allHighlights
+    ) throws TranslateException {
+        Encoding encodings = translator.getTokenizer().encode(question, context);
+        Encoding[] overflowEncodings = encodings.getOverflowing();
```

Comment:
Noticed that in `predictSentenceHighlightingQA()`, we're sending the full context in every overflow chunk and relying on chunk_number for the model/translator to figure out which part to process.

This assumes the translator re-chunks internally in the same way as the tokenizer, which could break silently if that logic ever changes.

Suggest either:

Add a comment clarifying that translator is expected to handle chunk slicing based on chunk_number, or

Pass actual sliced context from overflowEncodings[i] to make it explicit and safer.